tests/unit-test.el

```diff
@@ -9,3 +9,67 @@
   (it "returns the ns name of its argument"
     (expect (cljr--ns-name "com.corp.foo") :to-equal "foo")
     (expect (cljr--ns-name "foo") :to-equal "foo")))
+
+(defmacro with-clojure-file (filename &rest body)
+  (declare (indent 1))
+  `(with-temp-buffer
+     (setq-local buffer-file-name ,filename)
+     (delay-mode-hooks (clojure-mode))
+     ,@body))
+
+(describe "cljr--insert-require-libspec"
+  (it "adds a new require"
+    (with-clojure-file "libspec.clj"
+      (insert "(ns lib)\n")
+      (cljr--insert-require-libspec "[com.a :as a]")
+      (expect (buffer-string) :to-equal
+              "(ns lib\n  (:require [com.a :as a]))\n")
+      (cljr--insert-require-libspec "[com.b :as b]")
+      (expect (buffer-string) :to-equal
+              "(ns lib
+  (:require [com.a :as a]
+            [com.b :as b]))
+"))))
+
+(describe "cljr--ns-alias-at-point"
+  (it "returns the short alias before the /"
+    (with-temp-buffer
+      (insert "aba/")
+      (expect (cljr--ns-alias-at-point) :to-equal "aba")))
+
+  (it "removes namespace keyword ::"
+    (with-temp-buffer
+      (insert "::ns-name/")
+      (expect (cljr--ns-alias-at-point) :to-equal "ns-name")))
+
+  (it "removes deref operator"
+    (with-temp-buffer
+      (insert "@atom/")
+      (expect (cljr--ns-alias-at-point) :to-equal "atom"))))
+
+(describe "cljr--unresolved-alias-ref"
+  (it "returns unresolved alias reference"
+    (with-clojure-file "foo.clj"
+      (insert "(ns foo)")
+      (expect (cljr--unresolved-alias-ref "ns-name")
+              :to-equal "ns-name")))
+
+  (it "returns nil for resolved alias"
+    (with-clojure-file "foo.clj"
+      (insert "(ns foo (:require [user.ns-name :as ns-name]))")
+      (expect (cljr--unresolved-alias-ref "ns-name") :to-be nil)))
+
+  (it "returns nil for js alias in cljs file."
+    (with-clojure-file "foo.cljs"
+      (insert "(ns foo)")
+      (expect (cljr--unresolved-alias-ref "js") :to-be nil)))
+
+  (it "returns js for js alias in clj file."
+    (with-clojure-file "foo.clj"
+      (insert "(ns foo)")
+      (expect (cljr--unresolved-alias-ref "js") :to-equal "js")))
+
+  (it "returns js for js alias in cljc file."
+    (with-clojure-file "foo.cljc"
+      (insert "(ns foo)")
+      (expect (cljr--unresolved-alias-ref "js") :to-equal "js"))))
```

Comment:
I think I was modeling this on https://github.com/clojure-emacs/clojure-mode/blob/master/test/clojure-mode-font-lock-test.el#L35, https://github.com/clojure-emacs/cider/blob/master/test/cider-font-lock-tests.el#L32, https://github.com/clojure-emacs/cider/blob/master/test/cider-overlay-tests.el#L25, and https://github.com/clojure-emacs/cider/blob/master/test/cider-util-tests.el#L34, which generally seem to wrap the macro outside of the expect. I'm happy to follow your suggestion, but that's why I made that style assumption. 

I was seeing test failures locally inside of the macro during development but I could see how it's possible it could be swallowed in the macro. I guess if it's outside it also disallows any expectations based on the current buffer state? Ie you couldn't assert against the value of `(point)` or the current active region outside of the context of the temp buffer in the expected value. That said, that might not be the best way to make those assertions anyway.

Happy to change it if that's your style preference, just wanted to give the context.

Reply:
Oh, I just went to fix this and I noticed https://github.com/clojure-emacs/clj-refactor.el/pull/522/files#diff-f78a6cdcd522a7fae3dbadcc142377d69570a8088e0fcb28c4f6d51ae0f7b8fcR22-R32. If the expect needs to be on the outside I guess that means I need to split the test into two expectations one for each step. I feel it's clearer with multiple examples to keep context outside (and avoids a bunch of test setup duplication), but can also appreciate if you would prefer to have one expectation per test. Any preferences here?

Reply:
I'd like sth like the following:

```lisp
(describe "cljr--insert-require-libspec"
  (let* ((after-inserting-a)
         (after-inserting-b))
    (with-clojure-file "libspec.clj"
      (insert "(ns lib)\n")
      (cljr--insert-require-libspec "[com.a :as a]")
      (setq after-inserting-a (buffer-string))
      (cljr--insert-require-libspec "[com.b :as b]")
      (setq after-inserting-b (buffer-string))))
  (it "adds a new require, over an empty ns form"
    (expect after-inserting-a :to-equal "..."))
  (it "adds a new require, over a ns form with existing requires"
    (expect after-inserting-b :to-equal "...")))
```